### ISSUE TITLE
Confusing order-dependency in command-line arguments

### ISSUE THREAD

**Issue by nchammas** [docker/machine#632]
```
$ docker-machine -v
docker-machine version 0.1.0
```

Here's an example:

```
$ docker-machine create --driver vmwarefusion --vmwarefusion-memory-size 2048 vmware
INFO[0000] Creating SSH key...                          
INFO[0000] Creating VM...                               
INFO[0003] Waiting for VM to come online...             
INFO[0042] "vmware" has been created and is now the active machine. 
INFO[0042] To point your Docker client at it, run this in your shell: $(docker-machine env vmware) 
$ docker-machine inspect vmware
{
    ...
        "Memory": 2048,
    ...
}
$ docker-machine rm vmware
```

Everything here worked as expected.

Now, if I move the `--vmwarefusion-memory-size` option to the end:

```
$ docker-machine create --driver vmwarefusion vmware --vmwarefusion-memory-size 2048
INFO[0000] Creating SSH key...                          
INFO[0000] Creating VM...                               
INFO[0002] Waiting for VM to come online...             
INFO[0040] "vmware" has been created and is now the active machine. 
INFO[0040] To point your Docker client at it, run this in your shell: $(docker-machine env vmware) 
$ docker-machine inspect vmware
{
    ...
        "Memory": 1024,
    ...
}
$ docker-machine rm vmware
```

:open_mouth:

I was super confused at first, since this was the first invocation I actually tried.

I am sure there is some subtlety here, but can we make the options somehow position-independent?

Alternately, can we warn the user clearly if a specified option is being ignored?

**Comment by ehazlett** [comment]
Yeah unfortunately this is a limitation of how we are using an upstream cli lib.   We could probably check for args after the machine name.

/cc @sthulb @nathanleclaire

**Comment by sthulb** [comment]
I'm writing my own CLI lib, perhaps we can use it a bit later.

**Comment by nathanleclaire** [comment]
Sorry to hear it was confusing @nchammas.  Perhaps there is a way we can update the documentation to make the syntax / argument order more clear?  I'd be -1 on parsing flags after arguments are specified - that's not a common pattern.  For instance, `docker run -it ubuntu bash -v /tmp/foo:/foo` won't work and IMO shouldn't.

Interestingly, I tried this on the console out of curiosity (`git grep -i foo` will do a case-insensitive search for `foo` in a git repository):

```console
$ git grep foo -i
fatal: bad flag '-i' used after filename
```

I'm a fan of this approach, though I am not sure how if/how it could work in `codegangsta/cli` - perhaps we could contribute a patch upstream to enable a similar feature.

**Comment by nchammas** [comment]
Throwing an error sounds like the right thing to do. Silently ignoring an option seems scary, because you'll think an option took when it didn't.

FWIW, the flow that led to my tacking on the option at the end was:

1. I started off with:

    ```
    docker-machine create --driver vmwarefusion vmware
    ```

2. I realized I needed more memory so I killed this machine.
3. I then scrolled up through my shell history to my most recent `create` command and tacked on the `--vmwarefusion-memory-size` option.
